server/manager/src/main/java/org/apache/accumulo/manager/tableOps/bulkVer2/LoadFiles.java

```diff
@@ -338,11 +338,15 @@ private long loadFiles(TableId tableId, Path bulkDir, LoadMappingIterator loadMa
         TabletsMetadata.builder(manager.getContext()).forTable(tableId).overlapping(startRow, null)
             .checkConsistency().fetch(PREV_ROW, LOCATION, LOADED, TIME).build()) {
 
+      // The tablet iterator and load mapping iterator are both iterating over data that is sorted
+      // in the same way. The two iterators are each independently advanced to find common points in
+      // the sorted data.
```

Comment:
`BulkSerialize` writes a file based on a sorted map. But, when it's read back in it's assuming that the file contents are still sorted. I'm just wondering if there is a test or something to validate the assumption that both inputs are sorted the same way. Is there a possibility that an admin could "fix" the bulk import file under some failure condition that would cause it to not be sorted? Should we re-sort after reading it back in?

Reply:
> Is there a possibility that an admin could "fix" the bulk import file under some failure condition that would cause it to not be sorted? Should we re-sort after reading it back in?

There is a check in PrepBulkImport.validateLoadMapping() that runs in the initial fate step.  However subsequent fate steps do not validate.  We could add validation to the LoadMappingIterator that checks the extents it reads are in sorted order, that would probably be a simple change to make and could be done in 2.1.

Reply:
Opened #5045

Reply:
Re resorting, thinking validation would be better as resorting the data may hide a bug elsewhere in the code.